validator/lib/validateTileset.js

```diff
@@ -173,3 +184,103 @@ function validateTileHierarchy(root, tilesetDirectory) {
+                returnString = 'box [' + tileBV.box + '] is not within box [' + parentBV.box + ']';
+                return returnString;
+            } else {
+                return returnString;
+            }
+        } else if (defined(tileBV.sphere) && defined(parentBV.sphere)) {
+            // Sphere in Sphere
+            var transformed_tileSphere = getTransformedSphere(tileBV.sphere, tileTransform);
+            var transformed_parentSphere = getTransformedSphere(parentBV.sphere, parentTransform);
+            if (sphereInsideSphere(transformed_tileSphere, transformed_parentSphere) !== true) {
+                returnString = 'sphere [' + tileBV.sphere + '] is not within sphere [' + parentBV.sphere + ']';
+                return returnString;
+            } else {
+                return returnString;
+            }
+        } else if (defined(tileBV.region)&& defined(parentBV.region)) {
+            // Region in Region
+            // Region does not update with transform
+            var transformed_tileRegion = tileBV.region;
+            var transformed_parentRegion = parentBV.region;
+            if (regionInsideRegion(transformed_tileRegion, transformed_parentRegion) !== true) {
+                returnString = 'region [' + tileBV.region + '] is not within region [' + parentBV.region + ']';
+                return returnString;
+            } else {
+                return returnString;
+            }
+        } else if (defined(tileBV.box) && defined(parentBV.sphere)) {
+            // Box in Sphere
+            var transformed_tileBox = getTransformedBox(tileBV.box, tileTransform);
+            var transformed_parentSphere = getTransformedSphere(parentBV.sphere, parentTransform);
+            if (boxInsideSphere(transformed_tileBox, transformed_parentSphere) !== true) {
+                returnString = 'box [' + tileBV.box + '] is not within sphere [' + parentBV.sphere + ']';
+                return returnString;
+            } else {
+                return returnString;
+            }
+        } else if (defined(tileBV.sphere) && defined(parentBV.box)) {
+            // Sphere in Box
+            var transformed_tileSphere = getTransformedSphere(tileBV.sphere, tileTransform);
+            var transformed_parentBox = getTransformedBox(parentBV.box, parentTransform);
+            if (sphereInsideBox(transformed_tileSphere, transformed_parentBox) !== true) {
+                returnString = 'sphere [' + tileBV.sphere + '] is not within box [' + parentBV.box + ']';
+                return returnString;
+            } else {
+                return returnString;
+            }
+        } else {
+            // Add more test cases here!
+            return returnString;
+        }
+    } else {
+        return returnString;
+    }
+}
+
+function getTransformedBox(box, transform) {
+    var scratchMatrix = new Matrix3();
+    var scratchHalfAxes = new Matrix3();
+    var scratchCenter = new Cartesian3();
```

Comment:
Put scratches outside of the function. That way they are not allocation for every function call, but instead exist in global scope.